packages/assets/generate.ts

```diff
@@ -0,0 +1,62 @@
+const fs = require('fs');
+const path = require('path');
+
+interface Entry {
+  name: string;
+  isDirectory: () => boolean;
+}
+
+const getFiles = async (entry: Entry) => {
+  const originalEntries = await fs.promises.readdir(entry, {
+    withFileTypes: true,
+  });
+
+  const files = await Promise.all(
+    originalEntries.map((foundEntry: Entry) => {
```

Comment:
Om vi installerer `@types/node` (typedefinisjoner til interne node pakker) så ser vi at dette er av typen `Dirent`.
Jeg har ikke helt testet ut enda, men når jeg begynte å nøste litt opp i typene her så virker det som det er noe blanding mellom arrays og ikke arrays og etterfulgt av en flatten måte å gjøre det på med concat. Jeg syns det er tidvis litt vanskelig å følge logikken.

Selv om det teknisk sett gjør det tregere så kan koden her forenkles (gitt at man ikke trigger parallelle IO operasjoner men gjør den "lineær". Som sagt litt dårligere performance, men jeg tror ikke det er flaskehalsen her tbh):

```ts
async function getFiles(entry: string): Promise<string[]> {
  const originalEntries = await fs.readdir(entry, {
    withFileTypes: true,
  });

  let files: string[] = [];
  for (let foundEntry of originalEntries) {
    const res = path.resolve(entry, foundEntry.name);
    const result = foundEntry.isDirectory() ? await getFiles(res) : [res];
    files = files.concat(result);
  }
  return files;
}
```

Bare et forslag fra min side etter at jeg prøvde å nøste litt opp i typene her. Kan også skrive om et forslag til å bedre utnytte asynkronitet om ønskelig.

Reply:
Siden vi ikke bygger ting mer avansert enn nødvendig for å dekke behov som muligens oppstår i fremtiden, tenker jeg ovennevnte forslag er fint, da det er noe mer intuitivt enn opprinnelig forslag 👍